2008-08-26  Janis Johnson  <janis187@us.ibm.com>

	* g++.dg/ext/altivec-3.C: Move AltiVec code out of main.

2008-04-04  Janis Johnson  <janis187@us.ibm.com>

	* gcc.dg/var-expand3.c: Skip for powerpc-linux if not on AltiVec HW.

--- gcc/testsuite/g++.dg/ext/altivec-3.C	(revision 139609)
+++ gcc/testsuite/g++.dg/ext/altivec-3.C	(revision 139610)
@@ -120,16 +120,19 @@ void baz2 (int i, ... )
     CHECK_INVARIANT (vec_all_eq (vxi.v, vx_g.v));
 }
 
-int main(void)
+void main1(void)
 {
     CHECK_INVARIANT (sizeof(struct foo) == 8 && sizeof(struct vfoo) == 48);
 
-    altivec_check();
-
     bar(i_1, x_g, (short)i_2, (float)d_2, ld_1, (char)i_1, d_3);
     baz(i_1, v_g, i_1, vx_g, i_1, v2_g, i_1, vx2_g); 
     quux(i_1, v_g, v_g);
     baz2(i_1, vx_g);
-    
+}
+
+int main(void)
+{
+    altivec_check();
+    main1();
     return 0;
 }
--- gcc/testsuite/gcc.dg/var-expand3.c	(revision 133910)
+++ gcc/testsuite/gcc.dg/var-expand3.c	(revision 133911)
@@ -1,4 +1,4 @@
-/* { dg-do run { target { powerpc*-*-* && powerpc_altivec_ok } } }} */
+/* { dg-do run { target { powerpc*-*-* && vmx_hw } } }} */
 /* { dg-options "-O2 -funroll-loops -ffast-math -fvariable-expansion-in-unroller -maltivec -dL" } */
 
 #include "altivec.h"
